src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -242,15 +244,10 @@ impl<S: StateStore> StorageTable<S> {
         let serialized_pk =
             serialize_pk_with_vnode(&pk, &self.pk_serializer, self.compute_vnode_by_pk(&pk));
         assert!(pk.len() <= self.pk_indices.len());
-        let key_indices = (0..pk.len())
-            .into_iter()
-            .map(|index| self.pk_indices[index])
-            .collect_vec();
 
         let read_options = ReadOptions {
-            dist_key_hint: None,
-            check_bloom_filter: !self.dist_key_indices.is_empty()
-                && self.dist_key_indices == key_indices,
+            prefix_hint: None,
+            check_bloom_filter: self.prefix_hint_len != 0 && self.prefix_hint_len <= pk.len(),
```

Comment:
This looks suspicious to me. How is `prefix_hint_len` derived for storage table? Is it possible that `prefix_hint_len < pk.len()`? If yes, there can be a problem because we always use the full pk to check bloom filter for point get.